test/sparse/sparse.jl

```diff
@@ -1755,3 +1755,73 @@ let
     @test map!(f, X, A, B, C) == sparse(map!(f, fX, fA, fB, fC))
     @test_throws DimensionMismatch map!(f, X, A, B, spzeros(N, M - 1))
 end
+
+# Test that broadcast! does not allocate unnecessarily
+let
+    # broadcast! over a single sparse matrix falls back to map!, tested above
+    N, M = 10, 12
+    # test broadcast! implementation specialized for a pair of (input) sparse matrices
```

Comment:
Is there a similar test for combinations of sparse matrices and scalars, e.g. `A .* α` where `α` is a scalar?